sequence_processing_pipeline/tests/test_QCJob.py

```diff
@@ -1,9 +1,357 @@
+import shutil
 import unittest
+from os.path import join, abspath, exists, basename
+from functools import partial
+from sequence_processing_pipeline.QCJob import QCJob
+from sequence_processing_pipeline.PipelineError import PipelineError
+from os import makedirs
+from metapool import KLSampleSheet, validate_and_scrub_sample_sheet
+import logging
+
+logging.basicConfig(level=logging.DEBUG)
 
 
 class TestQCJob(unittest.TestCase):
-    def test_something(self):
-        pass
+    def setUp(self):
+        # adjustable test_root helps w/testing in different environments.
+        test_root = abspath('sequence_processing_pipeline')
+        self.path = partial(join, test_root, 'tests', 'data')
+        self.sample_sheet_path = self.path('good-sample-sheet.csv')
+        self.mmi_db_path = self.path('mmi.db')
+        self.sample_run_dir = self.path('MyRunDir')
+        self.project_list = ['NYU_BMS_Melanoma_13059', 'Feist_11661',
+                             'Gerwick_6123']
+        self.qiita_job_id = 'abcdabcdabcdabcdabcdabcdabcdabcd'
+        self.maxDiff = None
+
+        try:
+            shutil.rmtree(self.sample_run_dir)
+        except FileNotFoundError:
+            # sample_run_dir is created during testing. we don't mind if it
+            # doesn't already exist, but any leftovers from previous runs
+            # need to be removed.
+            pass
+
+        sheet = KLSampleSheet(self.sample_sheet_path)
+        valid_sheet = validate_and_scrub_sample_sheet(sheet)
+
+        sample_ids = []
+        for sample in valid_sheet.samples:
+            sample_ids.append((sample['Sample_ID'], sample['Sample_Project']))
+
+        self.sample_paths = []
+        fastq_path = partial(join, self.sample_run_dir, 'Data', 'Fastq')
+        for project_name in self.project_list:
+            sample_path = fastq_path(project_name)
+            makedirs(sample_path, exist_ok=True)
+            self.sample_paths.append(sample_path)
+
+            ids = [x[0] for x in
+                   filter(lambda c: c[1] == project_name, sample_ids)]
+
+            for id in ids:
```

Comment:
It looks like here the code is creating the files in the test folder like: `filtered_sequences/sample2_R2_.trimmed.fastq.gz` thus we shouldn't need them in the repo, is that correct? If yes, please remove them from the repo and add an entry in `.gitignore` so they are not added by mistake again.

Reply:
Actually those are static. The code above is to create the raw fastq.gz files seen in Data/Fastq. Although it would be better to have the trimmed fastqs be created on demand as well. I can make it a low-level issue.